### ISSUE TITLE
Main README with build steps

### ISSUE THREAD

**Issue by message** [linuxmint/mint-themes#278]
Hey guys. Can you please add README file with build prerequisites (sassc, inkscape, optipng) and correct build steps?
When I build Mint themes my results differ from results in `/usr/share/themes`. Even though I launch `generate-themes.py` and `update-variations.py All`. When I install generated for example Mint-Y-Orange theme it has some elements from base theme, e.g. close window cross is green, not orange. What are a proper way to build and install Mint themes? I want to take a see how 1.8.4 new colors looked.

**Comment by SebastJava** [comment]
Hi Dmitry,
I don’t think there is anything wrong with your method. I used the same. I believe the problem you are referring to is there in the genuine Mint-Y, with Cinnamon. Unless i accidentally messed something. I can’t be totally sure. Interestingly, i found some weird workaround, as follow.

**In cinnamon-settings themes:**
1. Set  Window borders to Mint-Y, Controls to Mint-Y, Desktop to Mint-Y-Dark.
2. Change Controls to Mint-Y-Aqua. See? The window close button is still green. That’s a bug.
3. Change Desktop to anything else. Check the window close button now: it changed to Aqua!

...It looks like changing the Desktop "wakes up" the Window borders and Controls combination...

P.S. I certainly agree this repo needs a README.md. Take a look at mine: https://github.com/SebastJava/mint-yz-theme

**Comment by SebastJava** [comment]
I think i can fix this bug. Just watch for my next Debian release in a few days. :)

**Comment by hsbasu** [comment]
> Hi Dmitry, @message
> I don’t think there is anything wrong with your method. I used the same. I believe the problem you are referring to is there in the genuine Mint-Y, with Cinnamon. Unless i accidentally messed something. I can’t be totally sure. Interestingly, i found some weird workaround, as follow.
> 
> **In cinnamon-settings themes:**
> 
>     1. Set  Window borders to Mint-Y, Controls to Mint-Y, Desktop to Mint-Y-Dark.
> 
>     2. Change Controls to Mint-Y-Aqua. See? The window close button is still green. That’s a bug.
> 
>     3. Change Desktop to anything else. Check the window close button now: it changed to Aqua!
> 
> 
> ...It looks like changing the Desktop "wakes up" the Window borders and Controls combination...
> 
> P.S. I certainly agree this repo needs a README.md. Take a look at mine: https://github.com/SebastJava/mint-yz-theme

I don't think it's a bug. When you change controls to `aqua` in step 2, you are actually modifying a gtk-based window like the one used in `webapp-manager`. If you look at `webapp-manager` window, the close button will be `aqua`. But other apps like `nemo` which uses `metacity` window decoration/border, you will get `green` close button because the window control is set to `Mint-Y` not `Mint-Y-aqua`. To change window controls on `metacity` decorated windows, you need to change the window borders to `Mint-Y-Aqua`.

**Comment by SebastJava** [comment]
@hsbasu Yes, that is true, i got your point. But there is also another problem with Linux Mint Cinnamon and all the Mint-Y themes colour variations.

Take a look into `/usr/share/themes/Mint-Y/metacity-1/metacity-theme-3.xml`
lines 26-28:
```
<constant name="C_button_close_bg_focused" value="gtk:custom(selected_bg_color,#92b372)" />
<constant name="C_button_close_bg_hover" value="shade/gtk:custom(selected_bg_color,#92b372)/1.14" />
<constant name="C_button_close_bg_active" value="shade/gtk:custom(selected_bg_color,#92b372)/0.8" />
```
Now, try to find `/usr/share/themes/Mint-Y-Aqua/metacity-1/metacity-theme-3.xml`... This file does not exist. So, each colour variation is using this same `/usr/share/themes/Mint-Y/metacity-1/metacity-theme-3.xml`... And the **metacity theme** depends on this `selected_bg_color` variable which comes from the GTK theme. So, when you change the "Controls" in the **Cinnamon Menu > Preferences > Themes**, this changes the `selected_bg_color` variable into the GTK theme, and thus the **metacity** window border changes accordingly, supposedly. The problem is... this doesn't work! That's a bug. I can't remember what i had to do to make it work. I think you have to logout/login or something... Eventually, the **metacity** window close button will be the same as the GTK theme colour, but after a logout/login or something like that... I don't remember.

I tried to explain this to the Mint team before, but i was unsuccessful. This is an old and complex story...

**Comment by hsbasu** [comment]
> @hsbasu Yes, that is true, i got your point. But there is also another problem with Linux Mint Cinnamon and all the Mint-Y themes colour variations.
> 
> Take a look into `/usr/share/themes/Mint-Y/metacity-1/metacity-theme-3.xml`
> lines 26-28:
> ```
> <constant name="C_button_close_bg_focused" value="gtk:custom(selected_bg_color,#92b372)" />
> <constant name="C_button_close_bg_hover" value="shade/gtk:custom(selected_bg_color,#92b372)/1.14" />
> <constant name="C_button_close_bg_active" value="shade/gtk:custom(selected_bg_color,#92b372)/0.8" />
> ```
> Now, try to find `/usr/share/themes/Mint-Y-Aqua/metacity-1/metacity-theme-3.xml`... This file does not exist. So, each colour variation is using this same `/usr/share/themes/Mint-Y/metacity-1/metacity-theme-3.xml`... And the **metacity theme** depends on this `selected_bg_color` variable which comes from the GTK theme.

I'm not sure, but the quoted three lines I guess declare the variables. It does not use the variable from other files.

**Comment by JosephMcc** [comment]
@hsbasu Those lines are using a variable from the gtk theme. And yes there is a bug of some sort that is preventing the color of the close button from updating properly. It's known but the cause isn't as of yet. You can force it to update for now in ways mentioned above or by doing a Cinnamon restart.

**Comment by JosephMcc** [comment]
As to the original post by @message, the correct way to build the themes is using dpkg-buildpackage in the top level mint-themes directory and then installing the debs that it creates. It's the same method used to build any deb package from our git sources.

**Comment by SebastJava** [comment]
@hsbasu 

> I'm not sure, but the quoted three lines I guess declare the variables. It does not use the variable from gtk themes.

No.
1. This `#92b372` is some sort of fallback, i guess. The first value, this `selected_bg_color`, is the one being used, normally.
2. This `selected_bg_color` comes from the GTK theme: https://github.com/linuxmint/mint-themes/blob/master/src/Mint-Y/gtk-3.0/sass/_colors.scss line 11.

**Simple demonstration:** (tested!)
1. Go to LinuxMint Cinnamon Menu > Preferences > Themes.
2. Try to change the **Window borders** (metacity theme) to Mint-Y-Aqua. You can't. You only have the choice between Mint-Y or Mint-Y-Dark. So, choose Mint-Y.
3. Change the **Controls** (GTK theme) to Mint-Y-Aqua. Look: the window close button is still this khaki green colour so-called "Mint".
4. Logout/login. See? Now your window close button is okay: Aqua.

I tried to explain this to the team before, but was unsuccessful...
Check my fork: https://github.com/SebastJava/mint-yz-theme Try it! Everything is better there...

**Comment by JosephMcc** [comment]
> I tried to explain this to the team before, but was unsuccessful...

As I mentioned above, this is a known issue just don't know the fix yet.

And yes the #92b372 is a fallback color. The way that works is gtk:custom(color_from_theme, fallback_color_if_not_found)
